src/utils/GradleUtil.ts

```diff
@@ -0,0 +1,120 @@
+import * as path from 'path';
+import * as fse from "fs-extra";
+import { settings } from 'cluster';
+
+/**
+ * Check a build.gradle file for the liberty-gradle-plugin
+ * Return true if the build.gradle contains applies the liberty plugin
+ * @param buildFile JS object representation of the build.gradle
+ */
+export function validGradleBuild(buildFile: any) {
+    if (buildFile !== undefined) {
+        if (buildFile.apply !== undefined && buildFile.buildscript !== undefined && buildFile.buildscript.dependencies !== undefined) {
+            // check that "apply plugin: 'liberty'" is specified in the build.gradle
+            var libertyPlugin = false;
+            for (var i = 0; i < buildFile.apply.length; i++) {
+                if (buildFile.apply[i] == "plugin: 'liberty'") {
+                    libertyPlugin = true;
+                }
+            }
+            if (libertyPlugin) {
+                for (var i = 0; i < buildFile.buildscript.dependencies.length; i++) {
+                    var dependency = buildFile.buildscript.dependencies[i];
+                    // check that group matches io.openliberty.tools and name matches liberty-gradle-plugin
+                    if (dependency.group == "io.openliberty.tools" && dependency.name == "liberty-gradle-plugin") {
+                        console.debug("Found liberty-gradle-plugin in the build.gradle");
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+    return false;
+}
+
+/**
+ * Get the name of a gradle project
+ * If a settings.gradle exists with a name specified, return name
+ * Else return the parent directory name of the build.gradle
+ * @param path build.gradle location
+ */
+export async function getGradleProjetName(gradlePath: string): Promise<string> {
+    var dirName = path.dirname(gradlePath);
+    var gradleSettings = getGradleSettings(gradlePath);
+    var label = path.basename(dirName);
+    if (fse.existsSync(gradleSettings)) {
+        // File exists in path
+        var g2js = require('gradle-to-js/lib/parser');
+        label = await g2js.parseFile(gradleSettings).then(function (representation: any) {
+            if (representation["rootProject.name"] !== undefined) {
+                return representation["rootProject.name"];
+            }
+        }).catch((err: any) => console.error("Unable to parse settings.gradle: " + gradleSettings + "; " + err));
+    }
+    return label;
+}
+
+/**
+ * Get settings.gradle associated with a build.gradle
+ * Look in same directory as build.gradle or a child-folder
```

Comment:
Can you verify whether `master` is supposed to be a child or sibling directory of the root dir?

Reply:
According to this documentation: https://docs.gradle.org/current/userguide/build_lifecycle.html#sec:initialization
The `master` directory has the same nesting level as the current dir.